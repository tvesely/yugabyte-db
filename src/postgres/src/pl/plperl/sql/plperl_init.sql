-- test plperl.on_plperl_init errors are fatal

-- Must load plperl before we can set on_plperl_init
LOAD 'plperl';

SET SESSION plperl.on_plperl_init = ' system("/nonesuch") ';

SHOW plperl.on_plperl_init;

DO $$ warn 42 $$ language plperl;
