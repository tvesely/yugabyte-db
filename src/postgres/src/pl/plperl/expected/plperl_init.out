-- test plperl.on_plperl_init errors are fatal
-- Must load plperl before we can set on_plperl_init
LOAD 'plperl';
SET SESSION plperl.on_plperl_init = ' system("/nonesuch") ';
SHOW plperl.on_plperl_init;
 plperl.on_plperl_init 
-----------------------
  system("/nonesuch") 
(1 row)

DO $$ warn 42 $$ language plperl;
ERROR:  'system' trapped by operation mask at line 2.
CONTEXT:  while executing plperl.on_plperl_init
PL/Perl anonymous code block
